zeebe/engine/src/main/java/io/camunda/zeebe/engine/processing/user/IdentitySetupInitializer.java

```diff
@@ -103,9 +112,88 @@ public TaskResult execute(final TaskResultBuilder taskResultBuilder) {
                       .setClaimValue(mapping.getClaimValue())
                       .setName(mapping.getMappingId());
               setupRecord.addMapping(mappingrecord);
+              setupRecord.addRoleMember(
+                  new RoleRecord()
+                      .setRoleId(ADMIN_ROLE_ID)
+                      .setEntityType(EntityType.MAPPING)
+                      .setEntityId(mapping.getMappingId()));
             });
 
+    setupRecord.setDefaultTenant(
+        new TenantRecord().setTenantId(DEFAULT_TENANT_ID).setName(DEFAULT_TENANT_NAME));
+
     taskResultBuilder.appendCommandRecord(IdentitySetupIntent.INITIALIZE, setupRecord);
     return taskResultBuilder.build();
   }
+
+  private static void setupAdminRole(final IdentitySetupRecord setupRecord) {
+    setupRecord.addRole(new RoleRecord().setRoleId(ADMIN_ROLE_ID).setName("Admin"));
+    for (final var resourceType : AuthorizationResourceType.values()) {
+      if (resourceType == AuthorizationResourceType.UNSPECIFIED) {
+        // We shouldn't add empty permissions for an unspecified resource type
+        continue;
+      }
+
+      setupRecord.addAuthorization(
+          new AuthorizationRecord()
+              .setOwnerType(AuthorizationOwnerType.ROLE)
+              .setOwnerId(ADMIN_ROLE_ID)
+              .setResourceType(resourceType)
+              .setResourceId(WILDCARD_PERMISSION)
+              .setPermissionTypes(resourceType.getSupportedPermissionTypes()));
+    }
+    setupRecord.addTenantMember(
+        new TenantRecord()
+            .setTenantId(DEFAULT_TENANT_ID)
+            .setEntityType(EntityType.ROLE)
+            .setEntityId(ADMIN_ROLE_ID));
+  }
+
+  private static void setupConnectorsRole(final IdentitySetupRecord setupRecord) {
+    final var connectorsRoleId = "connectors";
+    setupRecord.addRole(new RoleRecord().setRoleId(connectorsRoleId).setName("Connectors"));
```

Comment:
❓ I wonder if for visibility (and testing potentially) its worth lifting values like these (roleId, name) to static variables on the class level, at the moment we have a mix of both (for example the Tenant name is also declared) above

Reply:
I don't really like to do that unless it's needed. Much easier if you can see everything in the method where it's being used rather than jumping between static fields and the method :shrug: